group-coordinator/src/main/java/org/apache/kafka/coordinator/group/OffsetAndMetadata.java

```diff
@@ -92,30 +117,34 @@ public boolean equals(Object o) {
 
         OffsetAndMetadata that = (OffsetAndMetadata) o;
 
-        if (offset != that.offset) return false;
+        if (committedOffset != that.committedOffset) return false;
         if (commitTimestampMs != that.commitTimestampMs) return false;
-        if (!leaderEpoch.equals(that.leaderEpoch)) return false;
-        if (!metadata.equals(that.metadata)) return false;
-        return expireTimestampMs.equals(that.expireTimestampMs);
+        if (recordOffset != that.recordOffset) return false;
+        if (!Objects.equals(leaderEpoch, that.leaderEpoch)) return false;
```

Comment:
any reason why we changed from the .equals to Objects.equals?  I guess it is safer if one of them can be null

Reply:
IntelliJ auto-generated this one. We should ask it. :). I suppose that it uses Objects.equals for the reason you mentioned.

Reply:
I see. Is this using AI 👀

Reply:
I don't think so. There is a feature to auto-generate equals and hashCode. It has been around for a while now.

Reply:
huh. Guess I never used it.